components/date_picker/DatePicker.js

```diff
@@ -67,12 +76,14 @@ const factory = (Input, DatePickerDialog) => {
             className={classnames(this.props.theme.input, {[inputClassName]: inputClassName })}
             error={this.props.error}
             onMouseDown={this.handleInputMouseDown}
+            onKeyPress={this.handleInputKeyPress}
```

Comment:
It this actually working? If you spread `props` in the last line and provide `onKeyPress` is provided, I think it's going to override that `handleInputKeyPress` assignment. You should either extract that prop at line 68 or spread the `others` variables as the first prop passed down.